### PR TITLE
Add the FolderTree widget documentation

### PR DESCRIPTION
**Pull request by serhiyzhovnir**
## Purpose of this pull request

This pull request (PR) adds the documentation for the FolderTree widget.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget-folder-tree.html
- https://devdocs.magento.com/guides/v2.2/javascript-dev-guide/widgets/widget-folder-tree.html

## Links to Magento source code

- https://github.com/magento/magento2/blob/2.3/app/code/Magento/Cms/view/adminhtml/web/js/folder-tree.js
- https://github.com/magento/magento2/blob/2.2/app/code/Magento/Cms/view/adminhtml/web/js/folder-tree.js

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/javascript-dev-guide/widgets/widget-folder-tree.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+- [currentPath](#currentpath)
+- [root](#root)
+- [rootName](#rootname)
+- [tree](#tree)
+- [url](#url)
+
+### `currentPath`
+
+The array path of folders IDs to the current opened folder.
```
Comment (dobooth) [inline guides/v2.2/javascript-dev-guide/widgets/widget-folder-tree.md]: Please clarify this sentence. I do not know what it means.

File: guides/v2.2/javascript-dev-guide/widgets/widget-folder-tree.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+**Type**: String
+
+**Default value**: `'Root'`
+
+### `tree`
+
+The configuration for [jstree](https://www.jstree.com) widget.
+
+The FolderTree widget used the [jstree](https://www.jstree.com) widget to build a hierarchical tree with folders.
```
Comment (dobooth) [inline guides/v2.2/javascript-dev-guide/widgets/widget-folder-tree.md]: 'The FolderTree widget uses...'

File: guides/v2.2/javascript-dev-guide/widgets/widget-folder-tree.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
+---
+group: javascript-developer-guide
+title: FolderTree widget
+contributor_name: Atwix
+contributor_link: https://www.atwix.com/
+---
+
+The FolderTree [widget](https://glossary.magento.com/widget/) allows creating a dynamic hierarchical tree structure with folders.
+
```
Comment (dobooth) [inline guides/v2.2/javascript-dev-guide/widgets/widget-folder-tree.md]: Please explain what this widget does in more detail. What can you do with it? What is the point of a dynamic hierarchical tree structure?

**Comment by serhiyzhovnir** [comment]
Hi @dobooth @rogyar 
I've adjusted this PR, can you please, review it again?
Let me know if any questions or if something looks not clear.
Thank you!

**Comment by dobooth** [comment]
running tests

**Comment by dobooth** [comment]
running tests

**Pull request merged**